pkg/kubelet/kubelet.go

```diff
@@ -110,8 +110,8 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
-	// nodeStatusUpdateRetry specifies how many times kubelet retries when posting node status failed.
-	nodeStatusUpdateRetry = 5
+	// nodeStatusUpdateRetryBackoff is the amount of time that we are willing to spend retrying to connect to the apiserver.
+	nodeStatusUpdateRetryMaximumTime = 20 * time.Minute
```

Comment:
Is that necessarily a bad thing? If a node is on a network that is severed or starved in CPU or some other weird condition , it might be good to evacuate . In the meantime it would be able to retry in a few minutes to be reincarnated as a valid node again.